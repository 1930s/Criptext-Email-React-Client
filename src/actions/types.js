export const Thread = {
  ADD_BATCH: 'ADD_THREADS',
  SELECT: 'SELECT_THREAD'
};

export const User = {
  ADD_BATCH: 'ADD_USERS',
  ADD: 'ADD_USER'
};

export const Email = {
  ADD_BATCH: 'ADD_EMAILS'
};
