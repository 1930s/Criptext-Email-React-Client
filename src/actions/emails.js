import { Email } from './types';

export const addEmails = emails => {
  return {
    type: Email.ADD_BATCH,
    emails: emails
  };
};

export const loadEmails = () => {
  return dispatch => {
    return fetch('/emails.json')
      .then(response => {
        if (response.status === 200) {
          return response.json();
        }
        return Promise.reject(response.status);
      })
      .then(json => {
        let emails = {};
        json.emails.forEach(element => {
          emails[element.id] = element;
        });
        dispatch(addEmails(emails));
      })
      .catch(err => {
        console.log(err);
      });
  };
};
