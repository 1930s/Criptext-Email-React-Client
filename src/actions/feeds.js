import { Feed } from './types';

export const addFeeds = feeds => {
  return {
    type: Feed.ADD_BATCH,
    feeds: feeds
  };
};

export const selectFeed = feedId => {
  return {
    type: Feed.SELECT,
    selectedFeed: feedId
  };
};

export const loadFeeds = () => {
  return async dispatch => {
    try {
      const response = await fetch('/feeds.json');
      const json = await response.json();
      const feeds = json.feeds;
      dispatch(addFeeds(feeds));
    } catch (e) {
      console.log(e);
    }
  };
};
