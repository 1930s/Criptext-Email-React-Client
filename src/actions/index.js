import * as Types from './types';

export const addThreads = threads => {
  return {
    type: Types.Thread.ADD_BATCH,
    threads: threads
  };
};

export const addUsers = users => {
  return {
    type: Types.User.ADD_BATCH,
    users: users
  };
};

export const addUser = user => {
  return {
    type: Types.User.ADD,
    user: user
  };
};

export const selectThread = threadId => {
  return {
    type: Types.Thread.SELECT,
    selectedThread: threadId
  };
};
