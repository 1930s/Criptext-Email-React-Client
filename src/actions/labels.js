import { Label } from './types';
const inLabel = 10;

export const addLabels = labels => {
  return {
    type: Label.ADD_BATCH,
    labels: labels
  };
};

export const addLabel = label => {
  let labels = {
    inLabel: {
      id: inLabel,
      text: label
    }
  };
  return {
    type: Label.ADD_BATCH,
    labels: labels
  };
};

export const loadLabels = () => {
  return dispatch => {
    return fetch('/labels.json')
      .then(response => {
        if (response.status === 200) {
          return response.json();
        }
        return Promise.reject(response.status);
      })
      .then(json => {
        let labels = {};
        json.labels.forEach(element => {
          labels[element.id] = {
            id: element.id,
            text: element.text
          };
        });
        dispatch(addLabels(labels));
      })
      .catch(err => {
        console.log(err);
      });
  };
};
