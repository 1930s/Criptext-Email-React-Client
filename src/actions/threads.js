import * as Types from './types';

export const addThreads = threads => {
  return {
    type: Types.Thread.ADD_BATCH,
    threads: threads.sort((t1, t2) => {
      return t1.lastEmailDate <= t2.lastEmailDate;
    })
  };
};

export const selectThread = threadId => {
  return {
    type: Types.Thread.SELECT,
    selectedThread: threadId
  };
};

export const multiSelectThread = (threadId, value) => {
  return {
    type: Types.Thread.MULTISELECT,
    selectedThread: threadId,
    value: value
  };
};

export const filterThreadsByUnread = enabled => {
  return {
    type: Types.Thread.UNREAD_FILTER,
    enabled: enabled
  };
};

export const addLabel = (threadId, label) => {
  return {
    type: Types.Thread.ADD_LABEL,
    targetThread: threadId,
    label: label
  };
};

export const removeLabel = (threadId, label) => {
  return {
    type: Types.Thread.REMOVE_LABEL,
    targetThread: threadId,
    label: label
  };
};

export const removeThread = (threadId) => {
  return {
    type: Types.Thread.REMOVE,
    targetThread: threadId
  }
}

export const loadThreads = () => {
  return dispatch => {
    return fetch('/threads.json')
      .then(response => {
        if (response.status === 200) {
          return response.json();
        }
        return Promise.reject(response.status);
      })
      .then(json => {
        dispatch(addThreads(json.threads));
      })
      .catch(err => {
        console.error(err);
      });
  };
};
