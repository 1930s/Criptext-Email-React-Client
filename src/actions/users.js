import { User } from './types';

export const addUsers = users => {
  return {
    type: User.ADD_BATCH,
    users: users
  };
};

export const addUser = user => {
  return {
    type: User.ADD,
    user: user
  };
};
