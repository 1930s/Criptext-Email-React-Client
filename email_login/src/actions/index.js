import { addUser, addUsers } from './users';

export { addUser, addUsers };
