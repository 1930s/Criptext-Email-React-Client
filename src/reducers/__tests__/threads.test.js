/* eslint-env node, jest */

import threadsReducer from '../thread';
import * as actions from '../../actions/index';

describe('thread actions', () => {
  it('should add threads to state', () => {
    const threads = {
      '23': {
        id: 23,
        subject: 'test subject 23',
        preview: 'preview 23',
        date: 1234567890,
        emails: [10, 43]
      },
      '13': {
        id: 13,
        subject: 'test subject 13',
        preview: 'preview 13',
        date: 1234567891,
        emails: [11, 21]
      }
    };
    const action = actions.addThreads(threads);
    const state = threadsReducer(null, action);
    expect(state).toMatchSnapshot();
  });

  it('should set thread as read', () => {
    const threads = {
      '23': {
        id: 23,
        subject: 'test subject 23',
        preview: 'preview 23',
        date: 1234567890,
        emails: [10, 43]
      }
    };
    const action = actions.addThreads(threads);
    const state = threadsReducer(null, action);
    const actionSelect = actions.selectThread(23);
    const newState = threadsReducer(state, actionSelect);
    expect(newState).toMatchSnapshot();
  });
});
