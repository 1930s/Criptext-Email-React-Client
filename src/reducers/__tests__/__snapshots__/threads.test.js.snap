// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`thread actions should add threads to state 1`] = `
Object {
  "13": Object {
    "date": 1234567891,
    "emails": Array [
      11,
      21,
    ],
    "id": 13,
    "preview": "preview 13",
    "subject": "test subject 13",
  },
  "23": Object {
    "date": 1234567890,
    "emails": Array [
      10,
      43,
    ],
    "id": 23,
    "preview": "preview 23",
    "subject": "test subject 23",
  },
}
`;

exports[`thread actions should set thread as read 1`] = `
Object {
  "23": Object {
    "date": 1234567890,
    "emails": Array [
      10,
      43,
    ],
    "id": 23,
    "preview": "preview 23",
    "subject": "test subject 23",
    "unread": false,
  },
}
`;
